Automattic/woocommerce-payments
# Fix keyboard navigation for account balance tooltips

Fixes #6053

#### Changes proposed in this Pull Request

This PR fixes the tab-based keyboard navigation for the **Payments → Overview** account balance tooltips.

The `TooltipBase` component has been updated to accept a `parentElement` prop that replaces the `rootElement` variable, which now allows a custom parent element for rendering the tooltip content within. All instances of `ClickTooltip` will pass in a custom parent element to `TooltipBase` that ensures the `tabIndex` of the rendered tooltip content will be able to be tabbed to directly after focusing on the tooltip button element.

The following screen recording demonstrates navigating the account balance tooltips + content via keyboard, using `Tab`/`Shift+Tab`:

https://github.com/Automattic/woocommerce-payments/assets/3147296/0458119c-f77f-43dc-9709-76c8d636f69b


#### Testing instructions





* Visit **Payments → Overview**
* Use the `Tab` key to navigate to each account balance tooltip.
* Pressing the `Enter` key should hide/show each tooltip.
* If a tooltip is visible, using the `Tab` key should focus on any link element present within.
* Ensure tooltip interaction via click and touch (mobile) is not affected.
  * **Note** I've fixed a bug on mobile safari that was introduced by this PR: [comment below](https://github.com/Automattic/woocommerce-payments/pull/6252#issuecomment-1543245775).

-------------------


**Post merge**
